IPython/nbconvert/filters/highlight.py

```diff
@@ -113,7 +113,7 @@ def _pygments_highlight(source, output_formatter, language='ipython', metadata=N
 
     # If the cell uses a magic extension language,
     # use the magic language instead.
-    if language == 'ipython' \
+    if language in ('ipython', 'ipython3') \
```

Comment:
Referring to line 122 in this file, and to [this line](https://github.com/ipython/ipython/blob/master/IPython/kernel/zmq/ipkernel.py#L114), I'm pretty sure that 'ipython2' is a valid value that may be passed to this function, even if it isn't in the case you've tested.